### ISSUE TITLE
ClayTreeView: keyboard alternative for drag and drop

### ISSUE THREAD

**Issue by marcoscv-work** [liferay/clay#5146]
Right now we can drag and drop and it works great, but we don't have alternate keys so it can be considered not accessible.

**We would need:**

- A way to invoque the action of move (for example, use the dropdown menu to call 'move action', when pressed start the interaction). 
<img width="625" alt="image" src="https://user-images.githubusercontent.com/7963804/196483356-cd35e013-f703-4ab9-909c-00b8e57a4e0c.png">

- Use arrow keys and enter to determinate the movement, example (only when the movement started):
https://schne324.github.io/dragon-drop/demo/#nested
<img width="225" alt="image" src="https://user-images.githubusercontent.com/7963804/196475670-4e8ac7a2-7c6b-4961-ac82-eccdee5a8e3a.png">

**Comment by matuzalemsteles** [comment]
Oh good point about that, I think this shouldn't be a problem to implement with what we have today.

**Comment by marcoscv-work** [comment]
Just some points to add before starting working on this:

_From a functional point of view_:
- We need to make the same by keyboard than by mouse so we need a way to open nodes and drop off the node in different levels, is important to remember that this is an alternative for accessibility so the easiest technical solutions if finally do the same will ve valid, we could auto-open groups, or do it in two steps dropping off the level-1 item in a level-2 group and then go back to the item in the level-2 group and move it again in that level

_From a design point o view_:
- Node in movement will be represented by the node in a white box with a blue line to determinate in the same way than using mouse the direction and effect of dropping of on the place blue line is colliding

![image](https://user-images.githubusercontent.com/7963804/196697542-8c686bc1-4791-4fd8-bf05-12d1cc61fc27.png)

![image](https://user-images.githubusercontent.com/7963804/196698000-ee36e5de-a44f-47f1-9c4d-c3474145869c.png)

**Comment by matuzalemsteles** [comment]
Hey @marcoscv-work sorry to comment too late here, maintaining the design will make this a little more complicated to maintain, i'm not sure this helps much when the user is interacting with the DnD using just the keyboard.

I think moving the Drag Layer wouldn't move in pixels but clicking on the arrow for example moves directly to the next drop point. I'd like to hear more from @drakonux and @emiliano-cicero about this before I go ahead with it, it would also be nice to first have something on the design side and documented before going with it.

**Comment by marcoscv-work** [comment]
Already implemented in Portal by Echo, if you guys want to take a look go there: Page editor -> Add fragments.

Please, we shouldn't forget this request, if any other way Echo will never use our tree / drag n drop, etc :-(

**Comment by matuzalemsteles** [comment]
I'm working on it but it's still in development I'm going to move it to the next release so it doesn't block the current release.
